File: resume-repository/shopinglist/app/src/main/java/com/skilbox/shopinglist/MainActivity.java
package com.skilbox.shopinglist;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ListView;

public class MainActivity extends AppCompatActivity {

    ListView listView ;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        listView = findViewById(R.id.list_view);
        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                if (position == 0){
                    Intent intent = new Intent(MainActivity.this,ProductsActivity.class);
                    startActivity(intent);
                }
                if (position == 1){
                    Intent intent = new Intent(MainActivity.this,ClothesActivity.class);
                    startActivity(intent);
                }
                if (position == 2){
                    Intent intent = new Intent(MainActivity.this,ToolsActivity.class);
                    startActivity(intent);
                }
                if (position == 3){
                    Intent intent = new Intent(MainActivity.this,HouseholdGoodsActivity.class);
                    startActivity(intent);
                }
            }
        });
    }
}

File: resume-repository/PermissionsAndDate/18_Data_Time/app/src/main/java/com/skilbox/a18_data_time/MainActivity.kt
package com.skilbox.a18_data_time

import android.app.Activity
import android.app.AlertDialog
import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.google.android.gms.common.ConnectionResult
import com.google.android.gms.common.ConnectionResult.SERVICE_INVALID
import com.google.android.gms.common.ConnectionResult.SERVICE_VERSION_UPDATE_REQUIRED
import com.google.android.gms.common.GoogleApiAvailability
import com.google.android.gms.common.GoogleApiAvailability.GOOGLE_PLAY_SERVICES_VERSION_CODE
import com.google.android.gms.common.GooglePlayServicesUtil

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val status = GoogleApiAvailability.getInstance().isGooglePlayServicesAvailable(this)

        when (status){
            SERVICE_INVALID -> needGoogleService()
            SERVICE_VERSION_UPDATE_REQUIRED-> {
                Toast.makeText(this, "Please update your google play service", Toast.LENGTH_SHORT).show()
                GoogleApiAvailability.getInstance().getErrorDialog(this,status,GOOGLE_PLAY_SERVICES_VERSION_CODE).show()
            }
            else -> startApp()
        }

    }

    private fun startApp() {
        supportFragmentManager.beginTransaction()
                .replace(R.id.conteiner_intraction, DangerousPermissionFragmrnt())
                .commit()
    }

    private fun needGoogleService() {
        AlertDialog.Builder(this)
                .setTitle("Приложение не может быть загружено т.к. не найден Google Service")
                .setCancelable(true)
    }


}
